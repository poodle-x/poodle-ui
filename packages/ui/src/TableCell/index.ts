import { TableCell, TableCellProps } from "./TableCell";
import { StandardThemeConfig } from "../theme";

export * from "./TableCell";

export default TableCell;

export type TableCellStyleKeys = "root";

export interface TableCellThemeConfig
	extends StandardThemeConfig<TableCellStyleKeys, TableCellProps> {}
