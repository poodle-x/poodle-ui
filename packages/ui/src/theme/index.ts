import React from "react";
import {
	ThemeValue,
	SystemThemeConfig,
	BreakpointsThemeConfig,
	SpacingThemeConfig,
	ColorsThemeConfig,
	FontsThemeConfig,
	BordersThemeConfig,
	ShadowThemeConfig,
	LayoutsThemeConfig,
} from "@poodle/system";

import { CSSObjectSystem, Stylesheet } from "../styled";
import { BoxThemeConfig } from "../Box";
import { ButtonThemeConfig } from "../Button";
import { ColumnsThemeConfig } from "../Columns";
import { ColumnThemeConfig } from "../Column";
import { ContainerThemeConfig } from "../Container";
import { TableThemeConfig } from "../Table";
import { TableHeadThemeConfig } from "../TableHead";
import { TableBodyThemeConfig } from "../TableBody";
import { TableRowThemeConfig } from "../TableRow";
import { TableCellThemeConfig } from "../TableCell";
import { VisuallyHiddenThemeConfig } from "../VisuallyHidden";
import { CheckControlThemeConfig } from "../CheckControl";
import { CheckboxThemeConfig } from "../Checkbox";
import { RadioThemeConfig } from "../Radio";
import { InputThemeConfig } from "../Input";
import { InputAdornmentThemeConfig } from "../InputAdornment";
import { IconButtonThemeConfig } from "../IconButton";
import { ModalThemeConfig } from "../Modal";
import { ModalHeaderThemeConfig } from "../ModalHeader";
import { ModalBodyThemeConfig } from "../ModalBody";
import { ModalFooterThemeConfig } from "../ModalFooter";
import { SelectThemeConfig } from "../Select";
import { TabsThemeConfig } from "../Tabs";
import { TabThemeConfig } from "../Tab";
import { TabListThemeConfig } from "../TabList";
import { TabPanelThemeConfig } from "../TabPanel";

export { Stylesheet };

export { useTheme } from "./useTheme";

export type DirectionThemeConfig = "rtl" | "ltr";

export interface GlobalThemeConfig {
	styles?: ThemeValue<CSSObjectSystem>;
}

export type IconType = React.FC<
	React.PropsWithoutRef<React.SVGProps<SVGSVGElement>> &
		React.RefAttributes<SVGSVGElement>
>;

export interface BaseIcons {
	RadioButton?: IconType;
	RadioCheckedButton?: IconType;
	CheckboxButton?: IconType;
	CheckboxCheckedButton?: IconType;
	CheckboxIndeterminateButton?: IconType;
	ArrowDown?: IconType;
}

export interface ThemeConfig extends SystemThemeConfig {
	global?: GlobalThemeConfig;
	direction?: ThemeValue<DirectionThemeConfig>;
	baseIcons?: BaseIcons;
	Box?: BoxThemeConfig;
	Button?: ButtonThemeConfig;
	Columns?: ColumnsThemeConfig;
	Column?: ColumnThemeConfig;
	Container?: ContainerThemeConfig;
	Table?: TableThemeConfig;
	TableHead?: TableHeadThemeConfig;
	TableBody?: TableBodyThemeConfig;
	TableRow?: TableRowThemeConfig;
	TableCell?: TableCellThemeConfig;
	VisuallyHidden?: VisuallyHiddenThemeConfig;
	CheckControl?: CheckControlThemeConfig;
	Checkbox?: CheckboxThemeConfig;
	Radio?: RadioThemeConfig;
	Input?: InputThemeConfig;
	InputAdornment?: InputAdornmentThemeConfig;
	IconButton?: IconButtonThemeConfig;
	Modal?: ModalThemeConfig;
	ModalHeader?: ModalHeaderThemeConfig;
	ModalBody?: ModalBodyThemeConfig;
	ModalFooter?: ModalFooterThemeConfig;
	Select?: SelectThemeConfig;
	Tabs?: TabsThemeConfig;
	Tab?: TabThemeConfig;
	TabList?: TabListThemeConfig;
	TabPanel?: TabPanelThemeConfig;
}

export const defaultGlobal = (
	overrideTheme: ThemeConfig
): GlobalThemeConfig => {
	return {
		styles: overrideTheme.global?.styles,
	};
};

export const defaultBreakpoints = (
	overrideTheme: ThemeConfig
): BreakpointsThemeConfig => {
	return {
		scale: ["sm", "md", "lg", "xl"],
		...overrideTheme.breakpoints,
		sets: {
			sm: 576,
			md: 768,
			lg: 992,
			xl: 1200,
			xxl: 1400,
			...overrideTheme.breakpoints?.sets,
		},
	};
};

export const defaultSpacing = (
	overrideTheme: ThemeConfig
): SpacingThemeConfig => {
	return {
		scale: 4,
		...overrideTheme.spacing,
		sets: {
			s: 4,
			m: 8,
			l: 16,
			...overrideTheme.spacing?.sets,
		},
	};
};

export const defaultFonts = (overrideTheme: ThemeConfig): FontsThemeConfig => {
	return {
		scale: "1rem",
		...overrideTheme.fonts,
		base: {
			fontFamily:
				'-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"',
			fontStyle: "normal",
			fontWeight: "normal",
			lineHeight: "125%",
			fontSize: "1rem",
			...overrideTheme.fonts?.base,
		},
		weights: {
			normal: 400,
			semibold: 600,
			bold: 700,
			...overrideTheme.fonts?.weights,
		},
		sets: {
			input: {
				fontSize: "1rem",
				fontWeight: "normal",
				lineHeight: "125%",
			},
			button: {
				fontSize: "1rem",
				fontWeight: "bold",
				lineHeight: "150%",
			},
			label: {
				fontStyle: "normal",
				fontWeight: "bold",
				fontSize: "0.875rem",
				lineHeight: "110%",
			},
			helper: {
				fontStyle: "normal",
				fontWeight: "normal",
				fontSize: "0.875rem",
				lineHeight: "110%",
			},
			checkControlLabelMedium: {
				fontStyle: "normal",
				fontWeight: "normal",
				fontSize: "1rem",
				lineHeight: "110%",
			},
			checkControlLabelSmall: {
				fontStyle: "normal",
				fontWeight: "normal",
				fontSize: "0.875rem",
				lineHeight: "110%",
			},
			checkControlLabelLarge: {
				fontStyle: "normal",
				fontWeight: "normal",
				fontSize: "1.125rem",
				lineHeight: "110%",
			},
			modalTitle: {
				fontStyle: "normal",
				fontWeight: "bold",
				fontSize: "1.5rem",
				lineHeight: "133.33%",
			},
			...overrideTheme.fonts?.sets,
		},
	};
};

export const defaultColor = (overrideTheme: ThemeConfig): ColorsThemeConfig => {
	return {
		modes: {
			...overrideTheme.colors?.modes,
		},
		sets: {
			text: "#282828",
			disabledText: "#A6A6A6",

			primary: {
				light: "#D1EBFF",
				base: "#00478A",
				dark: "#003A70",
				textOnLight: "#00478A",
				textOnBase: "#fff",
				textOnDark: "#fff",
			},

			negative: {
				base: "#AF1308",
				dark: "#8D1007",
				light: "#FCF3F3",
				textOnLight: "#AF1308",
				textOnBase: "#fff",
				textOnDark: "#fff",
			},

			positive: {
				base: "#006122",
				dark: "#004D1B",
				light: "#E5FFEE",
				textOnLight: "#006122",
				textOnBase: "#fff",
				textOnDark: "#fff",
			},

			warn: {
				base: "#FFBB33",
				dark: "#FAAB00",
				light: "#FFF1DB",
				textOnLight: "#282828",
				textOnBase: "#282828",
				textOnDark: "#282828",
			},

			bg: {
				base: "#F8F8F8",
				100: "#E0E0E0",
				textOn100: "#282828",
				200: "#CCCCCC",
				textOn200: "#282828",
			},

			modal: {
				base: "#F8F8F8",
				footer: "#F0F0F0",
			},

			border: {
				base: "#CCCCCC",
			},

			...overrideTheme.colors?.sets,
		},
	};
};

export const defaultBorder = (
	overrideTheme: ThemeConfig
): BordersThemeConfig => {
	return {
		sets: {
			base: {
				style: "solid",
				width: "1px",
				color: "border",
			},
			...overrideTheme?.borders?.sets,
		},
		radius: {
			base: "4px",
			...overrideTheme?.borders?.radius,
		},
	};
};

export const defaultBoxShadow = (
	overrideTheme: ThemeConfig
): ShadowThemeConfig => {
	return {
		boxShadowSets: {
			lv1: "1px 2px 2px -1px rgba(0,0,0,0.4)",
			lv2: "2px 2px 5px 0px rgba(0,0,0,0.4)",
			lv3: "4px 4px 5px 0px rgba(0,0,0,0.4)",
			...overrideTheme?.shadows?.boxShadowSets,
		},
	};
};

export const defaultLayouts = (
	overrideTheme: ThemeConfig
): LayoutsThemeConfig => {
	return {
		zIndices: {
			modal: 2000,
			...overrideTheme?.layouts?.zIndices,
		},
	};
};

export function createTheme(overrideTheme: ThemeConfig = {}): ThemeConfig {
	return {
		direction: "ltr",
		...overrideTheme,
		global: defaultGlobal(overrideTheme),
		spacing: defaultSpacing(overrideTheme),
		breakpoints: defaultBreakpoints(overrideTheme),
		colors: defaultColor(overrideTheme),
		fonts: defaultFonts(overrideTheme),
		borders: defaultBorder(overrideTheme),
		shadows: defaultBoxShadow(overrideTheme),
		layouts: defaultLayouts(overrideTheme),
	};
}

export type StylesConfig<T extends string> = {
	[K in T]: ThemeValue<CSSObjectSystem>;
};

export interface StandardComponentProps {
	variant?: string;
	themeExtend?: ThemeConfig;
	theme?: ThemeConfig;
	sizeStyle?: string;
	colorStyle?: string;
}

export interface StandardThemeConfig<
	T extends string,
	Props extends { [key: string]: any }
> {
	defaultProps?: Partial<Props>;
	styles?: Partial<StylesConfig<T>>;
	variants?: {
		[key: string]: Partial<StylesConfig<T>>;
	};
	sizes?: {
		[key: string]: Partial<StylesConfig<T>>;
	};
	colorStyles?: {
		[key: string]: Partial<StylesConfig<T>>;
	};
	modes?: {
		[key: string]: {
			styles?: Partial<StylesConfig<T>>;
			variants?: {
				[key: string]: Partial<StylesConfig<T>>;
			};
			sizes?: {
				[key: string]: Partial<StylesConfig<T>>;
			};
			colorStyles?: {
				[key: string]: Partial<StylesConfig<T>>;
			};
		};
	};
	overrides?: Partial<StylesConfig<T>>;
	overrideClasses?: Partial<
		{
			[K in T]: boolean | string;
		}
	>;
}

export * from "./system";
