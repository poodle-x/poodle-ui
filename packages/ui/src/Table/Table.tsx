import React from "react";
import Box, { BoxProps } from "../Box";
import { StandardComponentProps, ThemeConfig } from "../theme";
import useDefaultProps from "../utils/useDefaultProps";
import { useClassNames } from "../styled";
import * as styles from "./styles";

const op = {};

export interface TableContextValue {
	withBorder?: boolean;
	withDivider?: boolean;
	withHover?: boolean;
	sizeStyle?: "s" | "m" | "l" | string;
}

export const TableContext = React.createContext<TableContextValue>(op);

export interface TableLevelContextValue {
	level?: "head" | "body" | "footer";
}

export const TableLevelContext = React.createContext<TableLevelContextValue>(
	op
);

export interface LocalTableProps {
	/**
	 * Apply root border.
	 */
	withBorder?: boolean;
	/**
	 * Add divider for each row.
	 */
	withDivider?: boolean;
	/**
	 * Apply hover style on table row.
	 */
	withHover?: boolean;
	/**
	 * Change table size. Built-in options are **"s" | "m" | "l"**.
	 *
	 * Default is **"m"**.
	 */
	sizeStyle?: "s" | "m" | "l" | string;
	/**
	 * Customize inner table props.
	 */
	tableProps?: BoxProps;
	children?: React.ReactNode;
}

export interface TableProps
	extends BoxProps,
		StandardComponentProps,
		LocalTableProps {}

function getDefaultProps(theme?: ThemeConfig) {
	return theme?.Table?.defaultProps;
}

export const Table: React.ForwardRefExoticComponent<
	React.PropsWithoutRef<TableProps> & React.RefAttributes<HTMLElement>
> = React.forwardRef<HTMLElement, TableProps>((_props, ref) => {
	const props = useDefaultProps<TableProps>(_props, {
		themeDefaultProps: getDefaultProps,
	});

	const {
		children,
		className,
		tableProps,
		withBorder,
		withDivider,
		withHover,
		sizeStyle = "m",
		...otherProps
	} = props;

	const classes = useClassNames({
		props,
		lists: {
			root: {
				classNames: ["poodle-table", styles.Root, className],
			},
			table: {
				classNames: [
					"poodle-table__table",
					styles.Table,
					tableProps?.className,
				],
			},
		},
	});

	return (
		<Box {...otherProps} className={classes.root} ref={ref}>
			<Box as="table" {...tableProps} className={classes.table}>
				<TableContext.Provider
					value={{
						sizeStyle,
						withBorder,
						withDivider,
						withHover,
					}}
				>
					{children}
				</TableContext.Provider>
			</Box>
		</Box>
	);
});

Table.defaultProps = {
	sizeStyle: "m",
};

Table.displayName = "PoodleTable";
