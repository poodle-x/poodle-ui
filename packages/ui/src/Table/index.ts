import { Table, TableProps } from "./Table";
import { StandardThemeConfig } from "../theme";

export * from "./Table";

export default Table;

export type TableStyleKeys = "root" | "table";

export interface TableThemeConfig
	extends StandardThemeConfig<TableStyleKeys, TableProps> {}
