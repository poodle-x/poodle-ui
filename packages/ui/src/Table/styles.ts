import { createCSSSystemStandard } from "../theme";
import { CSSSystem } from "../styled";
import { TableProps } from "./Table";
import { CSSObjectSystem } from "../styled";
import { TableStyleKeys } from "./";

export function Root(props: TableProps): CSSSystem {
	const { theme, withBorder, withHover } = props;

	const borderStyles: CSSObjectSystem = withBorder
		? {
				borderRadius: "base",
				border: "1px solid",
				borderColor: "border",

				"tr:last-child > td": {
					borderBottom: "0",
				},
		  }
		: {};

	const bgStyles: CSSObjectSystem = withHover
		? {
				"tbody > tr:hover": {
					bg: "bg.100",
					color: "bg.textOn100",
				},

				"tbody > tr[data-selected='true']": {
					bg: "primary.light",
					color: "primary.textOnLight",
				},
		  }
		: {};

	return createCSSSystemStandard<TableStyleKeys>({
		key: "root",
		config: theme?.Table,
		props,
		base: [borderStyles, bgStyles],
	});
}

export function Table(props: TableProps): CSSSystem {
	const { theme } = props;

	return createCSSSystemStandard<TableStyleKeys>({
		key: "table",
		config: theme?.Table,
		props,
		base: [
			{
				width: "100%",
				borderSpacing: 0,
				borderCollapse: "collapse",
				textAlign: "left",
			},
		],
	});
}
