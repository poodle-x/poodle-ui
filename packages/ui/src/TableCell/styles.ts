import { createCSSSystemStandard } from "../theme";
import { CSSSystem, CSSObjectSystem } from "../styled";
import { TableCellProps } from "./TableCell";
import { TableContextValue, TableLevelContextValue } from "../Table";
import { TableCellStyleKeys } from "./";

export function Root(
	props: TableCellProps & TableContextValue & TableLevelContextValue
): CSSSystem {
	const { theme, level, withDivider, sizeStyle = "" } = props;

	const levelTextStyles: CSSObjectSystem =
		level && level === "head"
			? {
					fontWeight: "bold",
			  }
			: {};

	const borderStyles: CSSObjectSystem =
		(level && level === "head") || withDivider
			? {
					borderBottom: "1px solid",
					borderBottomColor: "border",
			  }
			: {};

	let sizeStyles: CSSObjectSystem = {};

	switch (sizeStyle) {
		case "s": {
			sizeStyles = {
				py: "scale-2",
				px: "scale-4",
			};
			break;
		}
		case "l": {
			sizeStyles = {
				py: "scale-6",
				px: "scale-4",
			};
			break;
		}
		case "m":
		default: {
			sizeStyles = {
				py: "scale-4",
				px: "scale-4",
			};
		}
	}

	return createCSSSystemStandard<TableCellStyleKeys>({
		key: "root",
		config: theme?.TableCell,
		props,
		base: [borderStyles, levelTextStyles, sizeStyles],
	});
}
