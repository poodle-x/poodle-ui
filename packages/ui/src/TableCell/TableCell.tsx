import React from "react";
import Box, { BoxProps } from "../Box";
import { StandardComponentProps, ThemeConfig } from "../theme";
import useDefaultProps from "../utils/useDefaultProps";
import { useClassNames } from "../styled";
import * as styles from "./styles";
import { TableContext, TableLevelContext } from "../Table";

export interface LocalTableCellProps {
	children?: React.ReactNode;
}

export interface TableCellProps
	extends BoxProps,
		StandardComponentProps,
		LocalTableCellProps {}

function getDefaultProps(theme?: ThemeConfig) {
	return theme?.TableCell?.defaultProps;
}

export const TableCell: React.ForwardRefExoticComponent<
	React.PropsWithoutRef<TableCellProps> & React.RefAttributes<HTMLElement>
> = React.forwardRef<HTMLElement, TableCellProps>((_props, ref) => {
	const props = useDefaultProps<TableCellProps>(_props, {
		themeDefaultProps: getDefaultProps,
	});

	const tableContext = React.useContext(TableContext);
	const levelContext = React.useContext(TableLevelContext);

	const { children, className, ...otherProps } = props;

	const styleProps = React.useMemo(() => {
		return { ...props, level: levelContext.level, ...tableContext };
	}, [levelContext.level, props, tableContext]);

	const classes = useClassNames({
		props: styleProps,
		lists: {
			root: {
				classNames: ["poodle-table-cell", styles.Root, className],
			},
		},
	});

	return (
		<Box
			as={levelContext.level === "head" ? "th" : "td"}
			{...otherProps}
			className={classes.root}
			ref={ref}
		>
			{children}
		</Box>
	);
});

TableCell.displayName = "PoodleTableCell";
