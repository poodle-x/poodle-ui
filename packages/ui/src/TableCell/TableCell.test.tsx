import React from "react";
import { render } from "../utils/test";
import TableCell from "./";
import { ThemeConfig } from "../theme";
import { Table, TableContext, TableLevelContext } from "../Table";

describe("<TableCell />", () => {
	it("should render TableCell correctly", () => {
		const { container } = render(
			<table>
				<tbody>
					<tr>
						<TableCell>TableCell</TableCell>
					</tr>
				</tbody>
			</table>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("ref should work", () => {
		const ref = React.createRef<HTMLDivElement>();

		render(
			<table>
				<tbody>
					<tr>
						<TableCell ref={ref}>TableCell</TableCell>
					</tr>
				</tbody>
			</table>
		);
		expect(ref.current).toMatchSnapshot();
	});

	it("should render Box props correctly", () => {
		const { container } = render(
			<table>
				<tbody>
					<tr>
						<TableCell id="id" mt="20px">
							TableCell
						</TableCell>
					</tr>
				</tbody>
			</table>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("should render with level head correctly", () => {
		const { container } = render(
			<table>
				<TableLevelContext.Provider
					value={{
						level: "head",
					}}
				>
					<thead>
						<tr>
							<TableCell>TableCell</TableCell>
						</tr>
					</thead>
				</TableLevelContext.Provider>
			</table>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("should render with divider context correctly", () => {
		const { container } = render(
			<table>
				<TableContext.Provider
					value={{
						withDivider: true,
					}}
				>
					<tbody>
						<tr>
							<TableCell>TableCell</TableCell>
						</tr>
					</tbody>
				</TableContext.Provider>
			</table>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	describe("size", () => {
		it("should render with size s correctly", () => {
			const { container } = render(
				<Table sizeStyle="s">
					<tbody>
						<tr>
							<TableCell>TableCell</TableCell>
						</tr>
					</tbody>
				</Table>
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should render with size m correctly", () => {
			const { container } = render(
				<Table>
					<tbody>
						<tr>
							<TableCell>TableCell</TableCell>
						</tr>
					</tbody>
				</Table>
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should render with size l correctly", () => {
			const { container } = render(
				<Table sizeStyle="l">
					<tbody>
						<tr>
							<TableCell>TableCell</TableCell>
						</tr>
					</tbody>
				</Table>
			);
			expect(container.firstChild).toMatchSnapshot();
		});
	});

	describe("theming", () => {
		const themeTest: ThemeConfig = {
			mode: "mode",
			TableCell: {
				styles: {
					root: {
						mt: 10,
						letterSpacing: 250,
					},
				},
				variants: {
					custom: {
						root: {
							top: 0,
							textIndent: 1,
						},
					},
				},
			},
		};

		it("should extend styles in global", () => {
			const { container } = render(
				<table>
					<tbody>
						<tr>
							<TableCell variant="custom">TableCell</TableCell>
						</tr>
					</tbody>
				</table>,
				{
					theme: themeTest,
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should extend styles in local", () => {
			const { container } = render(
				<table>
					<tbody>
						<tr>
							<TableCell variant="custom" themeExtend={themeTest}>
								TableCell
							</TableCell>
						</tr>
					</tbody>
				</table>
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should render with default props correctly", () => {
			const { container } = render(
				<table>
					<tbody>
						<tr>
							<TableCell>TableCell</TableCell>
						</tr>
					</tbody>
				</table>,
				{
					theme: {
						TableCell: {
							defaultProps: {
								id: "id",
								className: "hello",
							},
						},
					},
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should disable classes", () => {
			const { container } = render(
				<table>
					<tbody>
						<tr>
							<TableCell>TableCell</TableCell>
						</tr>
					</tbody>
				</table>,
				{
					theme: {
						TableCell: {
							overrideClasses: {
								root: true,
							},
						},
					},
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should override classes", () => {
			const { container } = render(
				<table>
					<tbody>
						<tr>
							<TableCell>TableCell</TableCell>
						</tr>
					</tbody>
				</table>,
				{
					theme: {
						TableCell: {
							overrideClasses: {
								root: "morning",
							},
						},
					},
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should override styles", () => {
			const { container } = render(
				<table>
					<tbody>
						<tr>
							<TableCell>TableCell</TableCell>
						</tr>
					</tbody>
				</table>,
				{
					theme: {
						TableCell: {
							overrides: {
								root: {
									mt: 50,
								},
							},
						},
					},
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should extend size styles", () => {
			const { container } = render(
				<Table sizeStyle="test">
					<tbody>
						<tr>
							<TableCell>TableCell</TableCell>
						</tr>
					</tbody>
				</Table>,
				{
					theme: {
						TableCell: {
							sizes: {
								test: {
									root: {
										height: "100px",
									},
								},
							},
						},
					},
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should extend size styles with mode", () => {
			const { container } = render(
				<Table sizeStyle="test">
					<tbody>
						<tr>
							<TableCell>TableCell</TableCell>
						</tr>
					</tbody>
				</Table>,
				{
					theme: {
						mode: "test",
						TableCell: {
							sizes: {
								test: {
									root: {
										height: "100px",
									},
								},
							},
							modes: {
								test: {
									sizes: {
										test: {
											root: {
												height: "110px",
											},
										},
									},
								},
							},
						},
					},
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});
	});
});
