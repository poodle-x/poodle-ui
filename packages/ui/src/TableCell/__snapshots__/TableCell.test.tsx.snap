// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableCell /> ref should work 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
}

<td
  class="poodle-box c0 poodle-table-cell c1 c2"
>
  TableCell
</td>
`;

exports[`<TableCell /> should render Box props correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
}

.c2 {
  margin-top: 20px;
}

<table>
  <tbody>
    <tr>
      <td
        class="poodle-box c0 poodle-table-cell c1 c2"
        id="id"
      >
        TableCell
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<TableCell /> should render TableCell correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
}

<table>
  <tbody>
    <tr>
      <td
        class="poodle-box c0 poodle-table-cell c1 c2"
      >
        TableCell
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<TableCell /> should render with divider context correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  border-bottom: 1px solid;
  border-bottom-color: #CCCCCC;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
}

<table>
  <tbody>
    <tr>
      <td
        class="poodle-box c0 poodle-table-cell c1 c2"
      >
        TableCell
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<TableCell /> should render with level head correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  border-bottom: 1px solid;
  border-bottom-color: #CCCCCC;
  font-weight: 700;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
}

<table>
  <thead>
    <tr>
      <th
        class="poodle-box c0 poodle-table-cell c1 c2"
      >
        TableCell
      </th>
    </tr>
  </thead>
</table>
`;

exports[`<TableCell /> size should render with size l correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c4 {
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
  text-align: left;
}

.c7 {
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 16px;
  padding-right: 16px;
}

<div
  class="poodle-box c0 poodle-table c1 c2"
>
  <table
    class="poodle-box c0 poodle-table__table c4 c2"
  >
    <tbody>
      <tr>
        <td
          class="poodle-box c0 poodle-table-cell c7 c2"
        >
          TableCell
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<TableCell /> size should render with size m correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c4 {
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
  text-align: left;
}

.c7 {
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
}

<div
  class="poodle-box c0 poodle-table c1 c2"
>
  <table
    class="poodle-box c0 poodle-table__table c4 c2"
  >
    <tbody>
      <tr>
        <td
          class="poodle-box c0 poodle-table-cell c7 c2"
        >
          TableCell
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<TableCell /> size should render with size s correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c4 {
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
  text-align: left;
}

.c7 {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

<div
  class="poodle-box c0 poodle-table c1 c2"
>
  <table
    class="poodle-box c0 poodle-table__table c4 c2"
  >
    <tbody>
      <tr>
        <td
          class="poodle-box c0 poodle-table-cell c7 c2"
        >
          TableCell
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<TableCell /> theming should disable classes 1`] = `
.c0 {
  box-sizing: border-box;
}

<table>
  <tbody>
    <tr>
      <td
        class="poodle-box c0 poodle-table-cell c1"
      >
        TableCell
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<TableCell /> theming should extend size styles 1`] = `
.c0 {
  box-sizing: border-box;
}

.c4 {
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
  text-align: left;
}

.c7 {
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
  height: 100px;
}

<div
  class="poodle-box c0 poodle-table c1 c2"
>
  <table
    class="poodle-box c0 poodle-table__table c4 c2"
  >
    <tbody>
      <tr>
        <td
          class="poodle-box c0 poodle-table-cell c7 c2"
        >
          TableCell
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<TableCell /> theming should extend size styles with mode 1`] = `
.c0 {
  box-sizing: border-box;
}

.c4 {
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
  text-align: left;
}

.c7 {
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
  height: 100px;
  height: 110px;
}

<div
  class="poodle-box c0 poodle-table c1 c2"
>
  <table
    class="poodle-box c0 poodle-table__table c4 c2"
  >
    <tbody>
      <tr>
        <td
          class="poodle-box c0 poodle-table-cell c7 c2"
        >
          TableCell
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<TableCell /> theming should extend styles in global 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
}

<table>
  <tbody>
    <tr>
      <td
        class="poodle-box c0 poodle-table-cell c1 c2"
      >
        TableCell
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<TableCell /> theming should extend styles in local 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
}

<table>
  <tbody>
    <tr>
      <td
        class="poodle-box c0 poodle-table-cell c1 c2"
      >
        TableCell
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<TableCell /> theming should override classes 1`] = `
.c0 {
  box-sizing: border-box;
}

<table>
  <tbody>
    <tr>
      <td
        class="poodle-box c0 poodle-table-cell morning c1"
      >
        TableCell
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<TableCell /> theming should override styles 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  margin-top: 50px;
}

<table>
  <tbody>
    <tr>
      <td
        class="poodle-box c0 poodle-table-cell c1 c2"
      >
        TableCell
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<TableCell /> theming should render with default props correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
}

<table>
  <tbody>
    <tr>
      <td
        class="poodle-box c0 poodle-table-cell c1 hello c2"
        id="id"
      >
        TableCell
      </td>
    </tr>
  </tbody>
</table>
`;
